src/App.tsx

```diff
@@ -1,61 +1,137 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Post } from './types/Post';
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const initialPost = todosFromServer.map(post => ({
+  ...post,
+  user: getUserById(post.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setUserIdError] = useState(false);
+
+  function handleTitleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const cleanedValue = event.target.value
+      .replace(/[^a-zA-Zа-яА-Я0-9\s]/g, '');
+
+    setTitle(cleanedValue);
+    setHasTitleError(false);
+  }
+
+  function handleUserChange(event: React.ChangeEvent<HTMLSelectElement>) {
+    setUserId(+event.target.value);
+    setUserIdError(false);
+  }
+
+  function getNewPostId(posts: Post[]) {
+    const maxId = Math.max(...posts.map(post => post.id));
+
+    return maxId + 1;
+  }
+
+  const [todos, setPosts] = useState<Post[]>(initialPost);
```

Comment:
It's better to declare all states at the top level of the component body. This can give a clear overview of the component's state.